### DISCUSSION TITLE
save subset of pyramid tiles from memory buffer to hard drive using vips_dzsave_buffer

### DISCUSSION THREAD

**Discussion by cwwei2809** [libvips/libvips#3001] [Q&A] [answered]
Hi,

I am using dzsave to generate pyramids. I pasted the code below. Firstly, I use vips_dzsave to save all the generated pyramids into the hard drive as tif images (lines 42-47), which works well. To save the hard drive usage, I use vips_dzsave_buffer, intending to keep the pyramids in the memory buffer and only save selected tile(s) into the hard drive.

Lines 39-40 tend to read the first image (256 x 256) from the buffer and save it as a tif image (x.tif). It seems that the image was not saved anywhere.

Here are my questions:

Did I use the vips_dzsave_buffer correctly?
Is there a function to select a specific tile? Or treat the buffer as an array and access subset tiles with start index and length?
Appreciate it if you can provide an example of using vips_dzsave_buffer and only working on a subset of the pyramid tiles.

Thank you.

![image](https://user-images.githubusercontent.com/68030195/184794440-d56813b1-cbaf-47f1-9a1a-0e9602cb7f7e.png)
![image](https://user-images.githubusercontent.com/68030195/184794459-4b5f8883-7c0d-4c2a-996d-6b51942c5653.png)

**Comment by jcupitt** [comment, marked as answer]
Hi @cwwei2809,

`dzsave_buffer` writes a ZIP into memory, so you'll need a python package that can parse a zipfile and extract the files you want.

**Reply by cwwei2809** [reply]
Thank @jcupitt for answering.
